src/cdomains/intDomain.ml

```diff
@@ -923,6 +923,461 @@ struct
+      if d <= a' then 
+        of_bool ik true
+      else
+        if a > d' then of_bool ik false else top_bool
+  
+  let gt ik x y = 
+    let res = le ik x y in
+    if is_true res then 
+      zero 
+    else
+      if is_false res then one else top_bool
+
+  let ge ik x y = 
+    let res = lt ik x y in
+    if is_true res then 
+      zero 
+    else 
+      if is_false res then one else top_bool
+  
+  let eq ik x y = match x, y with
+    | (a, b)::[], (c, d)::[] when (Ints_t.compare a b) == 0 && (Ints_t.compare c d) == 0 && (Ints_t.compare a c) == 0 -> 
+      one
+    | _ -> 
+      if is_bot (meet ik x y) then 
+        zero 
+      else 
+        top_bool
+
+  let ne ik x y = 
+    let res = eq ik x y in
+    if is_true res then 
+      zero 
+    else 
+      if is_false res then one else top_bool
+
+  let bitand ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.bitand ik)
+
+  let bitor ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.bitor ik)
+
+  let bitnot ik x = 
+    unary_op x (wrap_unary_interval_function Interval_functor.bitnot ik)
+
+  let bitxor ik x y = 
+     binary_op ik x y (wrap_binary_interval_function Interval_functor.bitxor ik)
+
+  let shift_left ik x y = 
+     binary_op ik x y (wrap_binary_interval_function Interval_functor.shift_left ik)
+
+  let shift_right ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.shift_right ik)
+
+  let lognot ik x = 
+    unary_op x (wrap_unary_interval_function Interval_functor.lognot ik)
+
+  let logand ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.logand ik)
+
+  let logor ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.logor ik)
+
+  let add ?no_ov ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.add ik)
+
+  let neg ?no_ov ik x = 
+    unary_op x (wrap_unary_interval_function Interval_functor.neg ik)
+
+  let sub ?no_ov ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.sub ik)
+
+  let mul ?no_ov (ik: ikind) (x: t) (y: t) : t = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.mul ik)
+
+  let div ?no_ov ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.div ik)
+
+  let rem ik x y = 
+    binary_op ik x y (wrap_binary_interval_function Interval_functor.rem ik)
+
+  let cast_to ?torg ?no_ov ik x = List.map (fun x -> norm ~cast:true ik (Some x)) x |> List.flatten
+
+  let leq_interval x y = match x, y with
+  | None, _ -> true
+  | Some _, None -> false
+  | Some (x1,x2), Some (y1,y2) -> Ints_t.compare x1 y1 >= 0 && Ints_t.compare x2 y2 <= 0
+
+  let join_interval ik x y = match x, y with
+  | None, z | z, None -> z
+  | Some (x1,x2), Some (y1,y2) -> Some (Ints_t.min x1 y1, Ints_t.max x2 y2)
+
+  let rec interval_sets_to_partitions (ik: ikind) (acc : (int_t * int_t) option) (xs: t) (ys: t)= 
```

Comment:
What does this functions compute? 
If the `acc` is always going to be `None` when this function is called from the outside, you may also hide the parameter, by defining the recursive function locally.